developer/src/tike/main/KeymanDeveloperOptions.pas

```diff
@@ -507,6 +480,39 @@ procedure TKeymanDeveloperOptions.Write;
   WriteServerConfigurationJson;
 end;
 
+function TKeymanDeveloperOptions.BackOffAndSaveJson(const Filename: string; const JSON: TJSONObject): Boolean;
+var
+  count: Integer;
+begin
+  count := 0;
+  repeat
+    try
+      SaveJSONToFile(Filename, JSON);
+      Exit(True);
+    except
+      on E:EOSError do
+      begin
+        if E.ErrorCode = ERROR_ACCESS_DENIED then
+        begin
+          // Back off and retry 0.5 sec later, it's probably another
+          // instance happening to read or write the settings file
+          Inc(count);
+          Sleep(500);
+        end
+        else
+          raise;
+      end
+      else
+        raise;
+    end;
+  until count >= Max_Retries;
+
+  if TKeymanSentryClient.Instance <> nil then
+    TKeymanSentryClient.Instance.ReportMessage('TKeymanDeveloperOptions: Failed to write '+Filename+' after 5 tries');
```

Comment:
Since we already have this as a const
```suggestion
    TKeymanSentryClient.Instance.ReportMessage('TKeymanDeveloperOptions: Failed to write '+Filename+' after '+Max_Retries+' tries');
```

Reply:
```suggestion
    TKeymanSentryClient.Instance.ReportMessage('TKeymanDeveloperOptions: Failed to write '+Filename+' after '+IntToStr(Max_Retries)+' tries');
```